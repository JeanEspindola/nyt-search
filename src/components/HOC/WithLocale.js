import React from 'react';
import PropTypes from 'prop-types';

const WithLocale = (Component) => {
  const connectContext = (props, context) => (
    <Component {...props} locale={context.locale} />
  );

  connectContext.contextTypes = {
    locale: PropTypes.object,
  };

  return connectContext;
};

export default WithLocale;
