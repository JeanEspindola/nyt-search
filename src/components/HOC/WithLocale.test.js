import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import WithLocale from './WithLocale';

describe('WithLocale', () => {
  let Component;
  let WrapperComponent;

  beforeEach(() => {
    Component = () => (
      <div>Component</div>
    );

    WrapperComponent = WithLocale(Component);
  });

  it('renders WithLocale without crashing', () => {
    const wrapper = shallow(<WrapperComponent />);
    expect(wrapper.length).toBeTruthy();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
