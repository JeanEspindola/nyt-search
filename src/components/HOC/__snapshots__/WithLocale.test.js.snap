// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithLocale renders WithLocale without crashing 1`] = `<Component />`;
